app/commands/fetch_datasource_expenses.py

```diff
@@ -80,11 +70,45 @@ async def fetch_datasources_for_organizations(
             organization.id,
             organization.name,
         )
+        filtered_datasources = filter_relevant_datasources(response_datasources)
+        async with lock:
```

Comment:
There is no need to lock, there's no other `await` in your task function that can give the loop to some other task, and even in that case, each task handle a given organization so only one task will handle the same dict entry (key).